src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -48,17 +45,43 @@ class SubscriberArgs(parser: ArgParser) {
 
 private val PROJECT_ID = ServiceOptions.getDefaultProjectId()
 
-internal class TaskReceiver(private val tasksDir: Path,
-                            private val callback: (message: String, md5sum: String) -> Unit
-) : MessageReceiver {
-    private val dockerProcessor = DockerProcessor()
-
+abstract class CompilerFleetMessageReceiver : MessageReceiver {
     override fun receiveMessage(message: PubsubMessage, consumer: AckReplyConsumer) {
         processMessage(message)
         consumer.ack()
     }
 
-    fun processMessage(message: PubsubMessage) {
+    abstract fun processMessage(message: PubsubMessage)
+}
+
+internal class TaskReceiver(val tasksDirectory: String,
+                            private val callback: (message: String, md5sum: String) -> Unit
+) : CompilerFleetMessageReceiver() {
+    private val dockerProcessor = DockerProcessor()
+    private val resultTopic = "rmarkdown-results"
```

Comment:
Please extract it to command line arguments

Reply:
Done